Library/Homebrew/livecheck/strategy.rb

```diff
@@ -80,6 +83,40 @@ def from_url(url, livecheck_strategy: nil, regex_provided: nil)
           (strategy.const_defined?(:PRIORITY) ? -strategy::PRIORITY : -DEFAULT_PRIORITY)
         end
       end
+
+      def self.page_headers(url)
+        @headers ||= {}
+
+        return @headers[url] if @headers.key?(url)
+
+        headers = {}
+
+        [:default, :browser].each do |user_agent|
+          stdout, _, status = curl_output(
+            "--fail", "--head", "--request", "GET", "--silent", "--location",
+            "--connect-timeout", 5, "--retry-max-time", 15, "--max-time", 10,
```

Comment:
We can improve this in a follow-up PR.

Reply:
I already have this worked out as part of using `debug: false` for `curl` here and I'm planning to push the changes (along with the others) once the related PRs are merged.

I've stripped the `curl` changes from #10066 and I'll be opening in another PR right after that's merged. The `curl` PR will also include the `retry_request` parameter, so we could use it here. If we don't agree on the idea of the `retry_request` in `curl`, I can simply drop it in favor of only merging the commit to add `debug` to `#curl_with_workarounds`.

Since the work for all this is already done and it's part of some required changes, I don't see any harm in taking care of it now. However, if you think that there's an argument for leaving `--retry` enabled, I'll drop it for now.